packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -2134,10 +2134,19 @@ class DefaultAssetBundle extends InheritedWidget {
   /// The bundle to use as a default.
   final AssetBundle bundle;
 
-  /// The bundle from the closest instance of this class that encloses the given context.
+  /// The bundle from the closest instance of this class that encloses
+  /// the given context.
+  ///
+  /// If there is no [DefaultAssetBundle] ancestor widget in the tree
+  /// at the given context, then this will return the [rootBundle].
+  /// The [rootBundle] does not automatically select images based on
+  /// the current device pixel ratio. To get an asset bundle that
+  /// automatically performs pixel-density-aware asset resolution, use
+  /// a [MaterialApp], [WidgetsApp], or [AssetVendor] widget, which
+  /// introduce a suitably-configured [DefaultAssetBundle] widget.
   static AssetBundle of(BuildContext context) {
     DefaultAssetBundle result = context.inheritFromWidgetOfExactType(DefaultAssetBundle);
-    return result?.bundle;
+    return result?.bundle ?? rootBundle;
```

Comment:
(e.g. we regularly namespaces dart:ui imports "as ui", and mojom imports "as mojom".)